Android/app/src/main/java/app/intra/sys/DnsActivityReader.java

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2018 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.sys;
+
+import java.util.Collection;
+
+/**
+ * Visitor interface for DnsQueryTracker's activity data.  This allows a class to get access
+ * to the query activity without making a copy, while maintaining DnsQueryTracker's
+ * synchronization.
+ */
+public interface DnsActivityReader {
```

Comment:
The "reader" part of the class is a bit confusing to me. What does it read? It seems to me that "track" or something similar could be clearer.
(Also, what do you mean by visitor interface?)

Reply:
OK, thanks for the comment.  I renamed this and the related components, and improved the comments.